cpp/src/arrow/compute/exec/key_compare.cc

```diff
@@ -276,8 +276,9 @@ void KeyCompare::CompareVarBinaryColumnToRow(
         uint64_t key_right = key_right_ptr[j];
         result_or |= key_left ^ key_right;
       }
-      uint64_t tail_mask = ~0ULL >> (64 - 8 * (length - j * 8));
-      uint64_t key_left = util::SafeLoad(key_left_ptr + j);
+      int32_t tail_length = length - j * 8;
+      uint64_t tail_mask = ~0ULL >> (64 - 8 * (tail_length));
```

Comment:
nit: you probably meant `64 - (8 * tail_length)`? no need for extra parens

Reply:
I find the parens spare me the need of looking up C++ operator precedence rules.

Reply:
Well I'm referring to the parents around tail_length, where there are no operators inside.

Reply:
The parents around `tail_length` are unnecessary, and they are removed.